api_v3/services/MediaService.php

```diff
@@ -1182,6 +1182,67 @@ private function isClipTrimFlow($resource)
 			&& $resource->operationAttributes[0] instanceof KalturaClipAttributes);
 	}
 
+	/**
+	 * Get volume map by entry id
+	 *
+	 * @action getVolumeMap
+	 * @param string $entryId Entry id
+	 * @return file
+	 * @throws KalturaErrors::ENTRY_ID_NOT_FOUND
+	 */
+	function getVolumeMapAction($entryId)
+	{
+		$dbEntry = entryPeer::retrieveByPKNoFilter($entryId);
+		if (!$dbEntry || $dbEntry->getType() != KalturaEntryType::MEDIA_CLIP)
+			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $entryId);
+
+		$supportedFlavor = myEntryUtils::getFlavorSupportedByPackagerForVolumeMap($entryId);
+		if(!$supportedFlavor)
+			throw new KalturaAPIException(KalturaErrors::SUPPORTED_FLAVOR_NOT_EXIST, $entryId);
+		$supportedFlavorId = $supportedFlavor->getId();
+
+		$packagerRetries = 3;
+		$content = null;
+		while ($packagerRetries && !$content)
+		{
+			$content = $this->retrieveLocalVolumeMapFromPackager($supportedFlavor);
+			if(!$content)
+				$packagerRetries--;
+		}
+		if(!$content)
+			throw new KalturaAPIException(KalturaErrors::RETRIEVE_VOLUME_MAP_FAILED, $entryId);
+
+		header("Content-Disposition: attachment; filename=".$entryId.'_'.$supportedFlavorId."_volumeMap.csv");
+		return new kRendererString($content, 'text/csv');
+	}
+
+	private function retrieveLocalVolumeMapFromPackager($flavorAsset)
+	{
+		$packagerVolumeMapUrlPattern = kConf::get('packager_local_volume_map_url', 'local', null);
+		if (!$packagerVolumeMapUrlPattern)
```

Comment:
If the parameter is not defined, better throw a relevant exception then return null , because the result will be an exception saying - RETRIEVE_VOLUME_MAP_FAILED